[10주차] kmk

BLINK-ONCE opened the pull request.

BLINK-ONCE committed:
docs: Create java basics/

BLINK-ONCE committed:
Apply .gitignore

BLINK-ONCE committed:
style: Make InputView receive inputs, etc.

Make InputView receive inputs. getWinnersName() in Winners
get winners name by stream

PandaHun commented on racingcar/src/main/java/racingcar/Application.java:
```diff
@@ -0,0 +1,26 @@
+        nameCheck = Validator.validateName(carNames);
+        ErrorCatcher.catchNameError(nameCheck);
+
+        if (nameCheck[0] == 0 && nameCheck[1] == 0) {
```
뭘 의미하는 거죠?

PandaHun commented on racingcar/src/main/java/racingcar/Application.java:
```diff
@@ -0,0 +1,26 @@
+        cycle = inputView.inputCycle();
+
+        nameCheck = Validator.validateName(carNames);
+        ErrorCatcher.catchNameError(nameCheck);
```
전형적인 절차지향코드 형태입니다.

객체지향을 살려서 코드를 구현해 주세요.

선언과 할당이 분리될 필요가 없습니다

PandaHun commented on racingcar/src/main/java/view/InputView.java:
```diff
@@ -0,0 +1,22 @@
+
+    public String[] inputCarNames(){
+        System.out.println("경주할 자동차 이름을 입력하세요.(이름은 쉼표(,) 기준으로 구분)");
+        return SCANNER.nextLine().split(",");
```
매직 넘버란 코드 안에 작성된 구체적인 숫자 혹은 문자열을 말합니다. 그리고 매직 넘버의 단점은 아래와 같습니다.

* 의미를 이해하기 어렵다.
* 복수의 장소에서 사용된다면 변경하기 힘들다.

그런 매직 넘버는 의미를 나타낼 수 있는 상수로 치환하여 코드의 가독성을 높여 보는 것은 어떨까요?

PandaHun commented on racingcar/src/main/java/racingcar/GameController.java:
```diff
@@ -0,0 +1,39 @@
+                .collect(Collectors.toList());
+        OutputView.printResultTitle();
+
+        for (int i = 0; i < cycle; i++) {
```
`for` -> `stream` 변경해주세요

PandaHun commented on racingcar/src/main/java/racingcar/GameController.java:
```diff
@@ -0,0 +1,39 @@
+    }
+
+    public List<Car> moveCar() {
+        for (Car car : cars) {
```
`for` -> `stream`

PandaHun commented on racingcar/src/main/java/racingcar/Winners.java:
```diff
@@ -0,0 +1,26 @@
+        List<String> winnerName = new ArrayList<>();
+
+        winnerName = cars.stream()
+                .filter(x -> x.getPosition() == maxPosition)
```
해당부분을 메서드로 추출할 수 있을것 같은데요?

PandaHun commented on racingcar/src/main/java/utils/ErrorCatcher.java:
```diff
@@ -0,0 +1,17 @@
+        if (nameErrors[1] == 1) {
+            System.out.println("[ERROR] 이름의 길이가 부적절합니다.");
+        }
+    }
```
누군가가 이 코드를 받는다면 이해할 수 있을까요?

PandaHun commented on racingcar/src/main/java/utils/Validator.java:
```diff
@@ -0,0 +1,47 @@
+
+        // index 0에 이름 중복 검사 결과
+        // index 1에 이름 길이 검사 결과
+        validationResult[0] = isNameOnly(carNames);
```
주석이 필요한 코드는 충분한 의미를 담지 못했다고 생각됩니다

PandaHun commented on racingcar/src/main/java/utils/Validator.java:
```diff
@@ -0,0 +1,47 @@
+                validationResult[1] = 1;
+                break;
+            }
+        }
```
indent, for 등 요구사항을 만족시켜주세요

PandaHun commented on racingcar/src/main/java/utils/Validator.java:
```diff
@@ -0,0 +1,47 @@
+                return 1;
+            }
+            nameCount.put(carNames[i], "이것은 존재한.");
+        }
```
indent, for 등 요구사항을 만족시켜주세요

PandaHun reviewed (changes requested):
1. build와 gradle폴더에 `gitignore`추가 부탁드려요.
2. 공통적으로 코드에 마지막 줄이 없는 경우가 있습니다
이 또한 컨벤션이니 맞춰주세요
3. 이번 주차의 실습 요구사항을 맞춰주세요

재리뷰 요청드립니다.

BLINK-ONCE committed:
Modify .gitinore

BLINK-ONCE committed:
Untrack build, gradle dir

BLINK-ONCE committed:
Ignore test files

BLINK-ONCE committed:
docs: Add todo.md

BLINK-ONCE committed:
refactor: Replace some method to stream

array for car name validation has been changed to primitive
type variables. and method for car name validation has been
changed stream form. I madd main method length short.

BLINK-ONCE committed:
style: Delete magic number

I replaced magic number, SCANNER to not magic number,
scanner.

BLINK-ONCE committed:
docs: Update todo.md

BLINK-ONCE committed:
style: Add white space in last line
